app/controllers/media_objects_controller.rb

```diff
@@ -318,7 +318,7 @@ def custom_update
   def index
     respond_to do |format|
       format.json {
-        paginate json: MediaObject.all
+        paginate json: MediaObject.all.select { |mo| can? :read, mo }
```

Comment:
I think this is the right idea but I think this might cause scalability issues.  `MediaObject.all` is an `ActiveFedora::Relation` and there is some magic that operates on them to do pagination efficiently by limiting the number of items being returned by the solr query.  I think this change would fetch back all MediaObjects (via `to_a`) then pass them to `select.`

I think the same might be able to be achieved using `accessible_by`.
```suggestion
        paginate json: MediaObject.accessible_by(current_ability, :read)
```
See https://github.com/samvera/hydra-head/blob/main/hydra-access-controls/lib/active_fedora/accessible_by.rb#L4

Reply:
Ah, ok.  That makes sense since Avalon does a bunch of special things for gated discovery beyond what is done in hydra-access_controls.  It would be nice if `accessible_by` just worked.  I wonder how hard it would be to do that or to chain onto it so the correct results are returned to `paginate`.  Let me see what I can find.

Reply:
My guess as to why `accessible_by` isn't working as we expect is that switch from using `#gated_discovery_filters` to `Hydra::SearchBuilder#gated_discovery_filters` dropped our Avalon-specific overrides.  @masaball Can you try overriding the `accessibly_by` method to use `::SearchBuilder` instead of `Hydra::SearchBuilder` and see if that fixes things?

Add something like this to either the `hydra_config.rb` initializer or make a new initializer:
```
ActiveFedora::QueryMethods.module_eval do
  extend ActiveSupport::Concern

  def accessible_by(ability, action = :index)
    permission_types = case action
      when :index then [:discover, :read, :edit]
      when :show, :read then [:read, :edit]
      when :update, :edit, :create, :new, :destroy then [:edit]
    end

    builder = ::SearchBuilder.new(nil).with_ability(ability).with_discovery_permissions(permission_types)
    filters = builder.send(:gated_discovery_filters).join(" OR ")
    spawn.where!(filters)
  end
end
```